### PR TITLE
Return `RuntimeVersion` of overwritten/substituted wasm binary

### PR DESCRIPTION
**Pull request by bkchr**
Before this pr we always have returned the `RuntimeVersion` of the on chain wasm binary, even if there was an overwrite/substitute. This pull requests fixes this by returning always the `RuntimeVersion` of the used wasm binary. Most of the time these runtime versions should be the same. However, we already had different `RuntimeVersion` which made the node not detect the correct runtime api version (which are part of the runtime version).

### CODE REVIEW

File: client/service/src/client/call_executor.rs [hunk 1]
```diff
@@ -84,34 +82,55 @@ where
...
+		} else if let Some(s) =
+			self.wasm_substitutes
+				.get(on_chain_version.spec_version, onchain_code.heap_pages, id)
+		{
+			log::debug!(target: "wasm_substitutes", "Using WASM substitute for block {:?}", id);
+			s
+		} else {
+			log::debug!(
+				target: "wasm_overrides",
+				"No WASM override available for block {id}, using onchain code",
```
Comment (melekes) [inline client/service/src/client/call_executor.rs:109]:
```suggestion
				"Neither WASM override nor substitute available for block {id}, using onchain code",
```

File: client/service/src/client/call_executor.rs [hunk 2]
```diff
@@ -420,8 +433,64 @@ mod tests {
...
+				genesis_block_builder,
+				None,
+				Box::new(TaskExecutor::new()),
+				None,
+				None,
+				client_config,
+			)
+			.expect("Creates a client");
+
+		let version = client.runtime_version_at(&BlockId::Number(0)).unwrap();
```
Comment (melekes) [inline client/service/src/client/call_executor.rs:492]: should we test override in a similar fashion?

**Review by melekes — COMMENTED** [review]
👍

File: client/service/src/client/call_executor.rs (hunk 2 above)
Comment (bkchr) [reply client/service/src/client/call_executor.rs:492]: I think it is fine. Should be the same code path.

**Review by melekes — APPROVED** [review]
👍 (code-wise; don't know enough to reason about business logic)

**Review by michalkucharczyk — APPROVED** [review]

**Pull request merged**